### PR TITLE
test(integrations): add ng 10 ivy

### PR DESCRIPTION
**Pull request by joaqcid**
## PR Type

What kind of change does this PR introduce?

```
[ ] Other... Please describe:
```
Integration test for Angular v10 (Ivy enabled)

## Does this PR introduce a breaking change?

```
[ ] No
```

### CODE REVIEW

**Comment by joaqcid** [comment]
- I added same basic tests as ng9 (ivy). 
- I tried to include a few more tests, but i got an error.
- included script integration:ng10ivy to compile and run tests from root.

File: integrations/hello-world-ng10-ivy/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Editor configuration, see https://editorconfig.org
+root = true
+
+[*]
```
Comment (arturovt) [inline integrations/hello-world-ng10-ivy/.editorconfig:4]: Not needed.

File: integrations/hello-world-ng10-ivy/.gitignore [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# dependencies
+/node_modules
+
+# profiling files
+chrome-profiler-events*.json
+speed-measure-plugin*.json
+
+# IDEs and editors
+/.idea
+.project
```
Comment (arturovt) [inline integrations/hello-world-ng10-ivy/.gitignore:20]: Not needed.

File: integrations/hello-world-ng10-ivy/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Integration test ng-10 with Ivy enabled
```
Comment (arturovt) [inline integrations/hello-world-ng10-ivy/README.md:1]: Not needed.

**Comment by arturovt** [comment]
> I tried to include a few more tests, but i got an error.

Umm? :thinking:

**Comment by joaqcid** [comment]
> > I tried to include a few more tests, but i got an error.
> 
> Umm? 🤔

i tried running the default `app.component.spec.ts` that gets generated with ng new but i got the following error

image

and if i add `import 'jest-preset-angular';`

i get the following

image

**Comment by splincode** [comment]
@joaqcid https://github.com/ngxs/store/pull/1633/files#diff-77f86b6e6a488ea46db06421d97eeb77L1

yes, it's not need now

**Comment by joaqcid** [comment]
@splincode then the issue is the one above 

image

**Comment by markwhitfeld** [comment]
@joaqcid Could you add those tests back in (the failing ones) so that we can see the state of this from the build server?

**Comment by joaqcid** [comment]
added

**Comment by markwhitfeld** [comment]
That's strange... it's not failing in CI.
Did you add that project's tests to CI for both CircleCI and Travis?

File: .circleci/config.yml [hunk 4]
```diff
@@ -166,6 +166,14 @@ jobs:
...
           command: yarn integration:ivy
 
+  integration_ng10_ivy_tests:
+    <<: *job_defaults
+    steps:
+      - *attach_workspace
+      - run:
+          name: Run integration tests for ng10 (Ivy) application
+          command: yarn integration:ng10ivy                
+
```
Comment (joaqcid) [inline .circleci/config.yml:176]: @markwhitfeld i added this line, is that enough?

Comment (markwhitfeld) [reply .circleci/config.yml:176]: It needs to be added in some other places too.
Search through the file for `integration_ng` to find them ;-)

**Comment by markwhitfeld** [comment]
@joaqcid @splincode @arturovt Does anyone have an idea on how to resolve this?

**Comment by splincode** [comment]
we need remove `jest-preset-angular` :)

**Comment by splincode** [comment]
@joaqcid please create issue in Angular repo about 

```
  ● AppComponent › should render title

    This constructor is not compatible with Angular Dependency Injection because its dependency at index 0 of the parameter list is invalid.
    This can happen if the dependency type is a primitive like a string or if an ancestor of this class is missing an Angular decorator.

    Please check that 1) the type for the parameter at index 0 is correct and 2) the correct Angular decorators are defined for this class and its ancestors.

```

**Pull request merged**